src/evict/evict_lru.c

```diff
@@ -1801,12 +1804,25 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
      */
     start = queue->evict_queue + *slotp;
     remaining_slots = max_entries - *slotp;
-    if (btree->evict_walk_progress >= btree->evict_walk_target) {
-        btree->evict_walk_target = __evict_walk_target(session);
+
+    /*
+     * For this handle, calculate the number of target pages to evict. If the number of target pages
+     * is zero, then simply set the dhandle target and progress to zero and return early from this
+     * function.
+     *
+     * If the progress has not met the previous target, continue using the previous target.
+     */
+    if ((target_pages = __evict_walk_target(session)) == 0 ||
```

Comment:
I'm having a bit of trouble tracking this since we seem to use `target_pages` as a temporary variable (it's immediately rewritten after this branch) while also assigning inside the if check. Could we split this out into two separate checks for readability?

Reply:
As per the discussion, let's leave it as it is.